Type errors with aesara 2.7.8

```
pymc/util.py:241: error: Key expression in dictionary comprehension has incompatible type "Hashable"; expected type "str"
pymc/aesaraf.py:171: error: Argument 1 to "ndim" has incompatible type "Union[int, ndarray[Any, Any], Sequence[Union[int, Variable[Any, Any]]], TensorVariable[Any, Any]]"; expected "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]"
pymc/aesaraf.py:175: error: Incompatible types in assignment (expression has type "Tuple[Union[int, ndarray[Any, Any], Sequence[Union[int, Variable[Any, Any]]], TensorVariable[Any, Any]]]", variable has type "Union[int, ndarray[Any, Any], Sequence[Union[int, Variable[Any, Any]]], TensorVariable[Any, Any]]")
pymc/aesaraf.py:189: error: Argument 1 to "tuple" has incompatible type "Union[int, ndarray[Any, Any], Sequence[Union[int, Variable[Any, Any]]], TensorVariable[Any, Any]]"; expected "Iterable[Any]"
pymc/aesaraf.py:193: error: Argument 1 has incompatible type "Union[int, ndarray[Any, Any], Sequence[Union[int, Variable[Any, Any]]], TensorVariable[Any, Any]]"; expected "Union[Variable[Any, Any], Sequence[Variable[Any, Any]], Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]]"
pymc/aesaraf.py:229: error: Incompatible return value type (got "Tuple[None, None]", expected "Tuple[TensorVariable[Any, Any], TensorVariable[Any, Any]]")
pymc/aesaraf.py:233: error: Incompatible return value type (got "Tuple[TensorVariable[Any, Any], Optional[Any]]", expected "Tuple[TensorVariable[Any, Any], TensorVariable[Any, Any]]")
pymc/aesaraf.py:235: error: Incompatible return value type (got "Tuple[None, None]", expected "Tuple[TensorVariable[Any, Any], TensorVariable[Any, Any]]")
pymc/aesaraf.py:299: error: Incompatible types in "yield from" (actual type "Union[TensorVariable[Any, Any], Tuple[Any, Optional[Iterable[Any]]]]", expected type "TensorVariable[Any, Any]")
pymc/aesaraf.py:338: error: Incompatible types in assignment (expression has type "Dict[Union[Apply[Any], Variable[Any, Any], Op], Union[Apply[Any], Variable[Any, Any], Op]]", variable has type "Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]")
pymc/aesaraf.py:338: error: Argument 2 to "clone_get_equiv" has incompatible type "Iterable[TensorVariable[Any, Any]]"; expected "Sequence[Variable[Any, Any]]"
pymc/aesaraf.py:338: error: Argument 5 to "clone_get_equiv" has incompatible type "Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]"; expected "Optional[Dict[Union[Apply[Any], Variable[Any, Any], Op], Union[Apply[Any], Variable[Any, Any], Op]]]"
pymc/aesaraf.py:341: error: Invalid index type "Variable[Any, Any]" for "Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]"; expected type "TensorVariable[Any, Any]"
pymc/aesaraf.py:348: error: Argument 1 to "list" has incompatible type "List[Variable[Any, Any]]"; expected "Iterable[TensorVariable[Any, Any]]"
pymc/aesaraf.py:350: error: Incompatible return value type (got "Tuple[Iterable[TensorVariable[Any, Any]], Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]]", expected "Tuple[TensorVariable[Any, Any], Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]]")
pymc/aesaraf.py:409: error: Argument 2 to "clone_get_equiv" has incompatible type "Iterable[TensorVariable[Any, Any]]"; expected "Sequence[Variable[Any, Any]]"
pymc/aesaraf.py:410: error: List comprehension has incompatible type List[Union[Apply[Any], Variable[Any, Any], Op]]; expected List[TensorVariable[Any, Any]]
pymc/aesaraf.py:414: error: Key expression in dictionary comprehension has incompatible type "Union[Apply[Any], Variable[Any, Any], Op]"; expected type "TensorVariable[Any, Any]"
pymc/aesaraf.py:414: error: Value expression in dictionary comprehension has incompatible type "Union[Apply[Any], Variable[Any, Any], Op]"; expected type "TensorVariable[Any, Any]"
pymc/aesaraf.py:417: error: Argument 2 to "replace_rvs_in_graphs" has incompatible type "Callable[[Any, Any], Any]"; expected "Callable[[TensorVariable[Any, Any]], Dict[TensorVariable[Any, Any], TensorVariable[Any, Any]]]"
pymc/aesaraf.py:660: error: Invalid index type "Optional[str]" for "Dict[str, ndarray[Any, Any]]"; expected type "str"
pymc/aesaraf.py:669: error: Invalid index type "Optional[str]" for "Dict[str, ndarray[Any, Any]]"; expected type "str"
pymc/aesaraf.py:676: error: Argument 1 to "clone_replace" has incompatible type "TensorVariable[Any, Any]"; expected "Collection[Variable[Any, Any]]"
pymc/aesaraf.py:676: note: "TensorVariable" is missing following "Collection" protocol members:
pymc/aesaraf.py:676: note:     __contains__, __len__
pymc/aesaraf.py:676: error: Argument 2 to "clone_replace" has incompatible type "Dict[TensorVariable[Any, Any], Any]"; expected "Union[Iterable[Tuple[Variable[Any, Any], Variable[Any, Any]]], Dict[Variable[Any, Any], Variable[Any, Any]], None]"
pymc/aesaraf.py:938: error: Argument 1 to "SeedSequence" has incompatible type "Union[int, Sequence[int], ndarray[Any, Any], SeedSequence, None]"; expected "Union[None, int, Sequence[int], Union[_SupportsArray[dtype[Union[bool_, integer[Any]]]], _NestedSequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]], bool, int, _NestedSequence[Union[bool, int]]]]"
pymc/aesaraf.py:1017: error: Argument 1 to "reseed_rngs" has incompatible type "dict_keys[Any, Any]"; expected "Sequence[SharedVariable]"
pymc/initial_point.py:180: error: Argument 2 to "zip" has incompatible type "List[Union[RandomState, Generator]]"; expected "Iterable[Variable[Any, Any]]"
pymc/initial_point.py:317: error: Incompatible return value type (got "List[Variable[Any, Any]]", expected "List[TensorVariable[Any, Any]]")
pymc/initial_point.py:321: error: Incompatible return value type (got "List[Variable[Any, Any]]", expected "List[TensorVariable[Any, Any]]")
pymc/distributions/dist_math.py:63: error: Incompatible types in assignment (expression has type "List[Iterable[Variable[Any, Any]]]", variable has type "Tuple[Iterable[Variable[Any, Any]], ...]")
pymc/distributions/dist_math.py:330: error: Cannot instantiate abstract class "I1e" with abstract attribute "c_code"
pymc/distributions/dist_math.py:350: error: Cannot instantiate abstract class "I0e" with abstract attribute "c_code"
pymc/model_graph.py:54: error: Item "Tuple[Any, ...]" of "Union[Any, Tuple[Any, Optional[Iterable[Any]]]]" has no attribute "name"
pymc/model_graph.py:57: error: Incompatible return value type (got "Set[str]", expected "Set[VarName]")
pymc/model_graph.py:87: error: List comprehension has incompatible type List[Optional[str]]; expected List[VarName]
pymc/printing.py:130: error: "Variable[Any, Any]" has no attribute "str_repr"
pymc/printing.py:175: error: Item "Tuple[Any, ...]" of "Union[Any, Tuple[Any, Optional[Iterable[Any]]]]" has no attribute "owner"
pymc/printing.py:177: error: Item "Tuple[Any, ...]" of "Union[Any, Tuple[Any, Optional[Iterable[Any]]]]" has no attribute "name"
pymc/printing.py:197: error: Item "TensorVariable[Any, Any]" of "Union[TensorVariable[Any, Any], Model]" has no attribute "str_repr"
pymc/model.py:732: error: List item 0 has incompatible type "Union[Variable[Any, Any], Sequence[Variable[Any, Any]]]"; expected "TensorVariable[Any, Any]"
pymc/model.py:757: error: Incompatible types in assignment (expression has type "Union[TensorVariable[Any, Any], List[TensorVariable[Any, Any]]]", variable has type "List[TensorVariable[Any, Any]]")
pymc/model.py:761: error: Need type annotation for "potential_logps" (hint: "potential_logps: List[<type>] = ...")
pymc/model.py:763: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "List[Any]")
pymc/model.py:767: error: No overload variant of "__setitem__" of "list" matches argument types "int", "TensorVariable[Any, Any]"
pymc/model.py:767: note: Possible overload variants:
pymc/model.py:767: note:     def __setitem__(self, SupportsIndex, None) -> None
pymc/model.py:767: note:     def __setitem__(self, slice, Iterable[None]) -> None
pymc/model.py:770: error: Incompatible return value type (got "List[None]", expected "Union[Variable[Any, Any], List[Variable[Any, Any]]]")
pymc/model.py:809: error: List item 0 has incompatible type "Union[Variable[Any, Any], Sequence[Variable[Any, Any]]]"; expected "Variable[Any, Any]"
pymc/model.py:854: error: List item 0 has incompatible type "Union[Variable[Any, Any], Sequence[Variable[Any, Any]]]"; expected "Variable[Any, Any]"
pymc/model.py:881: error: Unsupported left operand type for + ("Variable[Any, Any]")
pymc/model.py:895: error: Incompatible return value type (got "Union[Variable[Any, Any], List[Variable[Any, Any]]]", expected "Variable[Any, Any]")
pymc/model.py:909: error: Incompatible return value type (got "Union[Variable[Any, Any], List[Variable[Any, Any]]]", expected "Variable[Any, Any]")
pymc/model.py:922: error: Incompatible return value type (got "Union[Variable[Any, Any], List[Variable[Any, Any]]]", expected "Variable[Any, Any]")
pymc/model.py:1141: error: Argument 1 to "array_equal" has incompatible type "Optional[Tuple[Any, ...]]"; expected "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]"
pymc/model.py:1141: error: Argument 2 to "array_equal" has incompatible type "Optional[Tuple[Any, ...]]"; expected "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]"
pymc/model.py:1148: error: Argument 1 to "len" has incompatible type "Optional[Tuple[Any, ...]]"; expected "Sized"
pymc/model.py:1198: error: "Variable[Any, Any]" has no attribute "set_value"
pymc/model.py:1237: error: "Dict[str, Optional[Sequence[Any]]]" has no attribute "ndim"
pymc/model.py:1237: error: "SharedVariable" has no attribute "ndim"
pymc/model.py:1239: error: "SharedVariable" has no attribute "ndim"
pymc/model.py:1243: error: Invalid index type "Optional[str]" for "Dict[str, Variable[Any, Any]]"; expected type "str"
pymc/model.py:1245: error: "Dict[str, Optional[Sequence[Any]]]" has no attribute "shape"
pymc/model.py:1246: error: Argument 1 to "get" of "dict" has incompatible type "Optional[str]"; expected "str"
pymc/model.py:1247: error: Argument 1 to "get" of "dict" has incompatible type "Optional[str]"; expected "str"
pymc/model.py:1308: error: Argument 1 to "set_dim" of "Model" has incompatible type "Optional[str]"; expected "str"
pymc/model.py:1491: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "ndarray[Any, Any]")
pymc/model.py:1493: error: Argument "value_var" to "create_value_var" of "Model" has incompatible type "ndarray[Any, Any]"; expected "Optional[Variable[Any, Any]]"
pymc/model.py:1538: error: Incompatible return value type (got "Union[Variable[Any, Any], Any]", expected "TensorVariable[Any, Any]")
pymc/model.py:1639: error: Incompatible return value type (got "Callable[..., Union[ndarray[Any, Any], List[ndarray[Any, Any]]]]", expected "Union[PointFunc, Callable[[Sequence[ndarray[Any, Any]]], Sequence[ndarray[Any, Any]]]]")
pymc/model.py:1920: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "Tuple[Any, ...]")
pymc/distributions/timeseries.py:94: error: Value of type "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None]" is not indexable
pymc/distributions/timeseries.py:95: error: Value of type "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None]" is not indexable
pymc/distributions/timeseries.py:95: error: Unsupported operand types for - ("Variable[Any, Any]" and "int")
pymc/distributions/timeseries.py:95: error: Unsupported operand types for - ("ellipsis" and "int")
pymc/distributions/timeseries.py:95: note: Left operand is of type "Union[Any, int, Variable[Any, Any], ellipsis]"
pymc/distributions/timeseries.py:99: error: Value of type "Optional[Tuple[Union[str, ellipsis, None], ...]]" is not indexable
pymc/distributions/timeseries.py:101: error: Unsupported operand types for - ("Variable[Any, Any]" and "int")
pymc/distributions/timeseries.py:101: error: Value of type "Optional[Tuple[Union[str, ellipsis, None], ...]]" is not indexable
pymc/distributions/timeseries.py:101: error: Invalid index type "Union[str, ellipsis, None, Any]" for "Dict[str, Variable[Any, Any]]"; expected type "str"
pymc/distributions/timeseries.py:153: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/timeseries.py:153: note:      Superclass:
pymc/distributions/timeseries.py:153: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/timeseries.py:153: note:      Subclass:
pymc/distributions/timeseries.py:153: note:          @classmethod
pymc/distributions/timeseries.py:153: note:          def rng_fn(cls, rng: RandomState, mu: Union[ndarray[Any, Any], float], sigma: Union[ndarray[Any, Any], float], init_dist: Union[ndarray[Any, Any], float], steps: int, size: Tuple[int]) -> ndarray[Any, Any]
pymc/distributions/timeseries.py:211: error: Value of type "Union[ndarray[Any, Any], float]" is not indexable
pymc/distributions/timeseries.py:212: error: Value of type "Union[ndarray[Any, Any], float]" is not indexable
pymc/distributions/timeseries.py:252: error: Signature of "dist" incompatible with supertype "Distribution"
pymc/distributions/timeseries.py:252: note:      Superclass:
pymc/distributions/timeseries.py:252: note:          @classmethod
pymc/distributions/timeseries.py:252: note:          def dist(cls, dist_params: Any, *, shape: Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None] = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/timeseries.py:252: note:      Subclass:
pymc/distributions/timeseries.py:252: note:          @classmethod
pymc/distributions/timeseries.py:252: note:          def dist(cls, mu: Any = ..., sigma: Any = ..., *, init_dist: Any = ..., steps: Any = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/timeseries.py:304: error: Self argument missing for a non-static method (or an invalid type for self)
pymc/distributions/timeseries.py:314: error: Argument 1 to "logp" has incompatible type "Variable[Any, Any]"; expected "TensorVariable[Any, Any]"
pymc/distributions/timeseries.py:314: error: Value of type "Variable[Any, Any]" is not indexable
pymc/distributions/timeseries.py:317: error: Value of type "Variable[Any, Any]" is not indexable
pymc/distributions/timeseries.py:320: error: Value of type "Variable[Any, Any]" is not indexable
pymc/distributions/timeseries.py:344: error: "Node" has no attribute "inputs"
pymc/distributions/timeseries.py:344: error: "Node" has no attribute "outputs"
pymc/distributions/timeseries.py:498: error: Item "Variable[Any, Any]" of "Union[Variable[Any, Any], FunctionGraph]" has no attribute "outputs"
pymc/distributions/timeseries.py:638: error: Cannot determine type of "_moment"
pymc/distributions/multivariate.py:93: error: Dispatch type "SimplexContinuous" must be subtype of fallback function first argument "Op"
pymc/distributions/multivariate.py:1250: error: Dispatch type "_LKJCholeskyCov" must be subtype of fallback function first argument "Op"
pymc/distributions/multivariate.py:1594: error: Dispatch type "LKJCorr" must be subtype of fallback function first argument "Op"
pymc/distributions/multivariate.py:2041: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/multivariate.py:2041: note:      Superclass:
pymc/distributions/multivariate.py:2041: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/multivariate.py:2041: note:      Subclass:
pymc/distributions/multivariate.py:2041: note:          @classmethod
pymc/distributions/multivariate.py:2041: note:          def rng_fn(cls, rng: RandomState, mu: Any, W: Any, alpha: Any, tau: Any, size: Any) -> Any
pymc/distributions/mixture.py:49: error: "Node" has no attribute "inputs"
pymc/distributions/mixture.py:49: error: "Node" has no attribute "outputs"
pymc/distributions/mixture.py:401: error: Cannot determine type of "_moment"
pymc/distributions/distribution.py:181: error: Incompatible return value type (got "Tuple[Any, Union[Sequence[Optional[str]], Tuple[Union[str, ellipsis, None], ...], None], Any, Union[TensorVariable[Any, Any], Tuple[Union[int, Variable[Any, Any]], ...], None]]", expected "Tuple[Variable[Any, Any], Optional[Tuple[Union[str, ellipsis, None], ...]], Union[ndarray[Any, Any], Variable[Any, Any], None], Union[TensorVariable[Any, Any], Tuple[Union[int, Variable[Any, Any]], ...]]]")
pymc/distributions/distribution.py:188: error: Incompatible types in assignment (expression has type "None", variable has type "RandomVariable")
pymc/distributions/distribution.py:190: error: Incompatible return type for "__new__" (returns "TensorVariable[Any, Any]", but must return a subtype of "Distribution")
pymc/distributions/distribution.py:269: error: Argument "rv" to "change_rv_size" has incompatible type "Variable[Any, Any]"; expected "TensorVariable[Any, Any]"
pymc/distributions/distribution.py:271: error: "None" has no attribute "register_rv"
pymc/distributions/distribution.py:282: error: "Variable[Any, Any]" has no attribute "str_repr"
pymc/distributions/distribution.py:283: error: "Variable[Any, Any]" has no attribute "_repr_latex_"
pymc/distributions/distribution.py:287: error: "Variable[Any, Any]" has no attribute "logp"
pymc/distributions/distribution.py:288: error: "Variable[Any, Any]" has no attribute "logcdf"
pymc/distributions/distribution.py:289: error: "Variable[Any, Any]" has no attribute "random"
pymc/distributions/distribution.py:290: error: Incompatible return value type (got "Variable[Any, Any]", expected "TensorVariable[Any, Any]")
pymc/distributions/distribution.py:343: error: Argument 1 to "convert_shape" has incompatible type "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None]"; expected "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]]]"
pymc/distributions/distribution.py:396: error: Incompatible return type for "__new__" (returns "TensorVariable[Any, Any]", but must return a subtype of "SymbolicDistribution")
pymc/distributions/distribution.py:471: error: "Type[SymbolicDistribution]" has no attribute "change_size"
pymc/distributions/distribution.py:477: error: "None" has no attribute "register_rv"
pymc/distributions/distribution.py:487: error: "Variable[Any, Any]" has no attribute "str_repr"
pymc/distributions/distribution.py:488: error: "Variable[Any, Any]" has no attribute "_repr_latex_"
pymc/distributions/distribution.py:492: error: Incompatible return value type (got "Variable[Any, Any]", expected "TensorVariable[Any, Any]")
pymc/distributions/distribution.py:543: error: Argument 1 to "convert_shape" has incompatible type "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None]"; expected "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]]]"
pymc/distributions/distribution.py:544: error: Argument 1 to "convert_size" has incompatible type "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any]]], None]"; expected "Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any]]]]"
pymc/distributions/distribution.py:547: error: "Type[SymbolicDistribution]" has no attribute "ndim_supp"
pymc/distributions/distribution.py:551: error: "Type[SymbolicDistribution]" has no attribute "rv_op"
pymc/distributions/distribution.py:556: error: "Type[SymbolicDistribution]" has no attribute "change_size"
pymc/distributions/distribution.py:753: error: Incompatible types in assignment (expression has type "List[TensorVariable[Any, Any]]", variable has type "Tuple[Any, ...]")
pymc/distributions/continuous.py:152: error: Dispatch type "PositiveContinuous" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:157: error: Dispatch type "UnitContinuous" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:162: error: Dispatch type "CircularContinuous" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:167: error: Dispatch type "BoundedContinuous" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:300: error: Incompatible types in assignment (expression has type "Tuple[int, int]", base class "BoundedContinuous" defined the type as "Optional[List[int]]")
pymc/distributions/continuous.py:342: error: Dispatch type "Uniform" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:349: error: Need type annotation for "ndims_params" (hint: "ndims_params: List[<type>] = ...")
pymc/distributions/continuous.py:420: error: Need type annotation for "ndims_params" (hint: "ndims_params: List[<type>] = ...")
pymc/distributions/continuous.py:597: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:597: note:      Superclass:
pymc/distributions/continuous.py:597: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:597: note:      Subclass:
pymc/distributions/continuous.py:597: note:          @classmethod
pymc/distributions/continuous.py:597: note:          def rng_fn(cls, rng: RandomState, mu: Union[ndarray[Any, Any], float], sigma: Union[ndarray[Any, Any], float], lower: Union[ndarray[Any, Any], float], upper: Union[ndarray[Any, Any], float], size: Union[List[int], int, None]) -> ndarray[Any, Any]
pymc/distributions/continuous.py:697: error: Incompatible types in assignment (expression has type "Tuple[int, int]", base class "BoundedContinuous" defined the type as "Optional[List[int]]")
pymc/distributions/continuous.py:700: error: Signature of "dist" incompatible with supertype "Distribution"
pymc/distributions/continuous.py:700: note:      Superclass:
pymc/distributions/continuous.py:700: note:          @classmethod
pymc/distributions/continuous.py:700: note:          def dist(cls, dist_params: Any, *, shape: Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None] = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/continuous.py:700: note:      Subclass:
pymc/distributions/continuous.py:700: note:          @classmethod
pymc/distributions/continuous.py:700: note:          def dist(cls, mu: Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None] = ..., sigma: Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None] = ..., tau: Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None] = ..., lower: Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None] = ..., upper: Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None] = ..., *args: Any, **kwargs: Any) -> RandomVariable
pymc/distributions/continuous.py:711: error: Argument 1 to "as_tensor_variable" has incompatible type "Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None]"; expected "Union[Variable[Any, Any], Sequence[Variable[Any, Any]], Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]]"
pymc/distributions/continuous.py:712: error: Argument 1 to "as_tensor_variable" has incompatible type "Union[ndarray[Any, Any], int, float, TensorVariable[Any, Any], None]"; expected "Union[Variable[Any, Any], Sequence[Variable[Any, Any]], Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]]"
pymc/distributions/continuous.py:717: error: Incompatible return value type (got "TensorVariable[Any, Any]", expected "RandomVariable")
pymc/distributions/continuous.py:780: error: Argument 2 to "_logprob" has incompatible type "Tuple[TruncatedNormal]"; expected "Tuple[TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:780: error: Argument 3 to "_logprob" has incompatible type "None"; expected "TensorVariable[Any, Any]"
pymc/distributions/continuous.py:780: error: Argument 4 to "_logprob" has incompatible type "None"; expected "TensorVariable[Any, Any]"
pymc/distributions/continuous.py:780: error: Argument 5 to "_logprob" has incompatible type "None"; expected "TensorVariable[Any, Any]"
pymc/distributions/continuous.py:780: error: Argument 6 to "_logprob" has incompatible type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"; expected "TensorVariable[Any, Any]"
pymc/distributions/continuous.py:780: error: Argument 7 to "_logprob" has incompatible type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"; expected "TensorVariable[Any, Any]"
pymc/distributions/continuous.py:783: error: Unsupported operand types for <= ("float" and "TruncatedNormal")
pymc/distributions/continuous.py:783: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:785: error: Unsupported operand types for >= ("float" and "TruncatedNormal")
pymc/distributions/continuous.py:785: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:788: error: Argument 2 to "check_parameters" has incompatible type "*List[Union[bool, ndarray[Any, dtype[bool_]], Any]]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:792: error: Dispatch type "TruncatedNormal" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:913: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:913: note:      Superclass:
pymc/distributions/continuous.py:913: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:913: note:      Subclass:
pymc/distributions/continuous.py:913: note:          @classmethod
pymc/distributions/continuous.py:913: note:          def rng_fn(cls, rng: Any, mu: Any, lam: Any, alpha: Any, size: Any) -> ndarray[Any, Any]
pymc/distributions/continuous.py:1002: error: Signature of "dist" incompatible with supertype "Distribution"
pymc/distributions/continuous.py:1002: note:      Superclass:
pymc/distributions/continuous.py:1002: note:          @classmethod
pymc/distributions/continuous.py:1002: note:          def dist(cls, dist_params: Any, *, shape: Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None] = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/continuous.py:1002: note:      Subclass:
pymc/distributions/continuous.py:1002: note:          @classmethod
pymc/distributions/continuous.py:1002: note:          def dist(cls, mu: Union[float, ndarray[Any, Any], None] = ..., lam: Union[float, ndarray[Any, Any], None] = ..., phi: Union[float, ndarray[Any, Any], None] = ..., alpha: Union[float, ndarray[Any, Any]] = ..., *args: Any, **kwargs: Any) -> RandomVariable
pymc/distributions/continuous.py:1011: error: Argument 1 to "get_mu_lam_phi" of "Wald" has incompatible type "Union[float, ndarray[Any, Any], None]"; expected "Optional[float]"
pymc/distributions/continuous.py:1011: error: Argument 2 to "get_mu_lam_phi" of "Wald" has incompatible type "Union[float, ndarray[Any, Any], None]"; expected "Optional[float]"
pymc/distributions/continuous.py:1011: error: Argument 3 to "get_mu_lam_phi" of "Wald" has incompatible type "Union[float, ndarray[Any, Any], None]"; expected "Optional[float]"
pymc/distributions/continuous.py:1012: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Union[float, ndarray[Any, Any]]")
pymc/distributions/continuous.py:1013: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Union[float, ndarray[Any, Any], None]")
pymc/distributions/continuous.py:1014: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Union[float, ndarray[Any, Any], None]")
pymc/distributions/continuous.py:1016: error: Incompatible return value type (got "TensorVariable[Any, Any]", expected "RandomVariable")
pymc/distributions/continuous.py:1071: error: Unsupported operand types for - ("Wald" and "float")
pymc/distributions/continuous.py:1071: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:1084: error: Argument 2 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1085: error: Argument 3 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1086: error: Argument 4 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1116: error: Unsupported operand types for - ("Wald" and "float")
pymc/distributions/continuous.py:1116: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:1116: error: Incompatible types in assignment (expression has type "Union[float, Any]", variable has type "Wald")
pymc/distributions/continuous.py:1117: error: Unsupported operand types for / ("Wald" and "float")
pymc/distributions/continuous.py:1117: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:1119: error: Unsupported operand types for * ("Wald" and "float")
pymc/distributions/continuous.py:1119: note: Right operand is of type "Union[float, ndarray[Any, Any], TensorVariable[Any, Any]]"
pymc/distributions/continuous.py:1135: error: Argument 2 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1135: error: Argument 3 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1135: error: Argument 4 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:1141: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:1141: note:      Superclass:
pymc/distributions/continuous.py:1141: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:1141: note:      Subclass:
pymc/distributions/continuous.py:1141: note:          @classmethod
pymc/distributions/continuous.py:1141: note:          def rng_fn(cls, rng: Any, alpha: Any, beta: Any, size: Any) -> ndarray[Any, Any]
pymc/distributions/continuous.py:1291: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:1291: note:      Superclass:
pymc/distributions/continuous.py:1291: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:1291: note:      Subclass:
pymc/distributions/continuous.py:1291: note:          @classmethod
pymc/distributions/continuous.py:1291: note:          def rng_fn(cls, rng: Any, a: Any, b: Any, size: Any) -> ndarray[Any, Any]
pymc/distributions/continuous.py:1600: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:1600: note:      Superclass:
pymc/distributions/continuous.py:1600: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:1600: note:      Subclass:
pymc/distributions/continuous.py:1600: note:          @classmethod
pymc/distributions/continuous.py:1600: note:          def rng_fn(cls, rng: Any, b: Any, kappa: Any, mu: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:1805: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:1805: note:      Superclass:
pymc/distributions/continuous.py:1805: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:1805: note:      Subclass:
pymc/distributions/continuous.py:1805: note:          @classmethod
pymc/distributions/continuous.py:1805: note:          def rng_fn(cls, rng: Any, nu: Any, mu: Any, sigma: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:1990: error: Incompatible types in assignment (expression has type "Tuple[int, None]", base class "BoundedContinuous" defined the type as "Optional[List[int]]")
pymc/distributions/continuous.py:1993: error: Signature of "dist" incompatible with supertype "Distribution"
pymc/distributions/continuous.py:1993: note:      Superclass:
pymc/distributions/continuous.py:1993: note:          @classmethod
pymc/distributions/continuous.py:1993: note:          def dist(cls, dist_params: Any, *, shape: Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None] = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/continuous.py:1993: note:      Subclass:
pymc/distributions/continuous.py:1993: note:          @classmethod
pymc/distributions/continuous.py:1993: note:          def dist(cls, alpha: Optional[float] = ..., m: Optional[float] = ..., **kwargs: Any) -> RandomVariable
pymc/distributions/continuous.py:1994: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Optional[float]")
pymc/distributions/continuous.py:1995: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Optional[float]")
pymc/distributions/continuous.py:1997: error: Incompatible return value type (got "TensorVariable[Any, Any]", expected "RandomVariable")
pymc/distributions/continuous.py:2005: error: Self argument missing for a non-static method (or an invalid type for self)
pymc/distributions/continuous.py:2036: error: Argument 2 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:2036: error: Argument 3 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:2040: error: Dispatch type "Pareto" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:2525: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:2525: note:      Superclass:
pymc/distributions/continuous.py:2525: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:2525: note:      Subclass:
pymc/distributions/continuous.py:2525: note:          @classmethod
pymc/distributions/continuous.py:2525: note:          def rng_fn(cls, rng: Any, alpha: Any, beta: Any, size: Any) -> ndarray[Any, Any]
pymc/distributions/continuous.py:2627: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:2627: note:      Superclass:
pymc/distributions/continuous.py:2627: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:2627: note:      Subclass:
pymc/distributions/continuous.py:2627: note:          @classmethod
pymc/distributions/continuous.py:2627: note:          def rng_fn(cls, rng: Any, nu: Any, sigma: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:2748: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:2748: note:      Superclass:
pymc/distributions/continuous.py:2748: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:2748: note:      Subclass:
pymc/distributions/continuous.py:2748: note:          @classmethod
pymc/distributions/continuous.py:2748: note:          def rng_fn(cls, rng: Any, mu: Any, sigma: Any, nu: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:2979: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:2979: note:      Superclass:
pymc/distributions/continuous.py:2979: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:2979: note:      Subclass:
pymc/distributions/continuous.py:2979: note:          @classmethod
pymc/distributions/continuous.py:2979: note:          def rng_fn(cls, rng: Any, mu: Any, sigma: Any, alpha: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:3148: error: Incompatible types in assignment (expression has type "Tuple[int, int]", base class "BoundedContinuous" defined the type as "Optional[List[int]]")
pymc/distributions/continuous.py:3202: error: Dispatch type "Triangular" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:3258: error: Signature of "dist" incompatible with supertype "Distribution"
pymc/distributions/continuous.py:3258: note:      Superclass:
pymc/distributions/continuous.py:3258: note:          @classmethod
pymc/distributions/continuous.py:3258: note:          def dist(cls, dist_params: Any, *, shape: Union[int, TensorVariable[Any, Any], Sequence[Union[int, Variable[Any, Any], ellipsis]], None] = ..., **kwargs: Any) -> TensorVariable[Any, Any]
pymc/distributions/continuous.py:3258: note:      Subclass:
pymc/distributions/continuous.py:3258: note:          @classmethod
pymc/distributions/continuous.py:3258: note:          def dist(cls, mu: Optional[float] = ..., beta: Optional[float] = ..., **kwargs: Any) -> RandomVariable
pymc/distributions/continuous.py:3260: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Optional[float]")
pymc/distributions/continuous.py:3261: error: Incompatible types in assignment (expression has type "TensorVariable[Any, Any]", variable has type "Optional[float]")
pymc/distributions/continuous.py:3263: error: Incompatible return value type (got "TensorVariable[Any, Any]", expected "RandomVariable")
pymc/distributions/continuous.py:3274: error: Self argument missing for a non-static method (or an invalid type for self)
pymc/distributions/continuous.py:3295: error: Argument 2 to "check_parameters" has incompatible type "Union[bool, ndarray[Any, dtype[bool_]], Any]"; expected "Iterable[Variable[Any, Any]]"
pymc/distributions/continuous.py:3306: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:3306: note:      Superclass:
pymc/distributions/continuous.py:3306: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:3306: note:      Subclass:
pymc/distributions/continuous.py:3306: note:          @classmethod
pymc/distributions/continuous.py:3306: note:          def rng_fn(cls, rng: Any, b: Any, sigma: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:3532: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:3532: note:      Superclass:
pymc/distributions/continuous.py:3532: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:3532: note:      Subclass:
pymc/distributions/continuous.py:3532: note:          @classmethod
pymc/distributions/continuous.py:3532: note:          def rng_fn(cls, rng: Any, mu: Any, sigma: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:3655: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:3655: note:      Superclass:
pymc/distributions/continuous.py:3655: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:3655: note:      Subclass:
pymc/distributions/continuous.py:3655: note:          @classmethod
pymc/distributions/continuous.py:3655: note:          def rng_fn(cls, rng: Any, x: Any, pdf: Any, cdf: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:3772: error: Dispatch type "Interpolated" must be subtype of fallback function first argument "Op"
pymc/distributions/continuous.py:3788: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:3788: note:      Superclass:
pymc/distributions/continuous.py:3788: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:3788: note:      Subclass:
pymc/distributions/continuous.py:3788: note:          @classmethod
pymc/distributions/continuous.py:3788: note:          def rng_fn(cls, rng: Any, mu: Any, sigma: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/distributions/continuous.py:3914: error: Signature of "rng_fn" incompatible with supertype "RandomVariable"
pymc/distributions/continuous.py:3914: note:      Superclass:
pymc/distributions/continuous.py:3914: note:          def rng_fn(self, rng: Any, *args: Any, **kwargs: Any) -> Any
pymc/distributions/continuous.py:3914: note:      Subclass:
pymc/distributions/continuous.py:3914: note:          @classmethod
pymc/distributions/continuous.py:3914: note:          def rng_fn(cls, rng: Any, h: Any, z: Any, size: Any = ...) -> ndarray[Any, Any]
pymc/step_methods/arraystep.py:106: error: Self argument missing for a non-static method (or an invalid type for self)
pymc/step_methods/arraystep.py:149: error: Signature of "step" incompatible with supertype "BlockedStep"
pymc/step_methods/arraystep.py:149: note:      Superclass:
pymc/step_methods/arraystep.py:149: note:          def [StatsType] step(point, *args: Any, **kwargs: Any) -> Union[Dict[str, ndarray[Any, Any]], Tuple[Dict[str, ndarray[Any, Any]], StatsType]]
pymc/step_methods/arraystep.py:149: note:      Subclass:
pymc/step_methods/arraystep.py:149: note:          def step(self, point: Dict[str, ndarray[Any, Any]]) -> Any
pymc/step_methods/arraystep.py:153: error: Argument 1 to "append" of "list" has incompatible type "Dict[str, ndarray[Any, Any]]"; expected "Callable[[RaveledVars], Any]"
pymc/step_methods/arraystep.py:155: error: Key expression in dictionary comprehension has incompatible type "Optional[str]"; expected type "str"
pymc/step_methods/arraystep.py:155: error: Invalid index type "Optional[str]" for "Dict[str, ndarray[Any, Any]]"; expected type "str"
pymc/step_methods/arraystep.py:156: error: Need type annotation for "step_res"
pymc/step_methods/arraystep.py:156: error: Argument 2 to "astep" of "ArrayStep" has incompatible type "*List[Callable[[RaveledVars], Any]]"; expected "Dict[str, ndarray[Any, Any]]"
pymc/step_methods/metropolis.py:61: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:62: error: Item "None" of "Optional[Generator]" has no attribute "normal"
pymc/step_methods/metropolis.py:68: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:69: error: Item "None" of "Optional[Generator]" has no attribute "uniform"
pymc/step_methods/metropolis.py:75: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:76: error: Item "None" of "Optional[Generator]" has no attribute "standard_cauchy"
pymc/step_methods/metropolis.py:82: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:84: error: Item "None" of "Optional[Generator]" has no attribute "standard_exponential"
pymc/step_methods/metropolis.py:90: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:91: error: Item "None" of "Optional[Generator]" has no attribute "poisson"
pymc/step_methods/metropolis.py:104: error: Incompatible types in assignment (expression has type Module, variable has type "Optional[Generator]")
pymc/step_methods/metropolis.py:106: error: Item "None" of "Optional[Generator]" has no attribute "normal"
pymc/step_methods/metropolis.py:109: error: Item "None" of "Optional[Generator]" has no attribute "normal"
pymc/step_methods/metropolis.py:256: error: "RaveledVars" has no attribute "astype"
pymc/step_methods/metropolis.py:265: error: "RaveledVars" has no attribute "copy"
pymc/step_methods/metropolis.py:395: error: Signature of "astep" incompatible with supertype "ArrayStep"
pymc/step_methods/metropolis.py:395: note:      Superclass:
pymc/step_methods/metropolis.py:395: note:          def [StatsType] astep(self, apoint: RaveledVars, point: Dict[str, ndarray[Any, Any]], *args: Any) -> Union[RaveledVars, Tuple[RaveledVars, StatsType]]
pymc/step_methods/metropolis.py:395: note:      Subclass:
pymc/step_methods/metropolis.py:395: note:          def astep(self, q0: RaveledVars, logp: Any) -> Tuple[RaveledVars, List[Dict[str, Any]]]
pymc/step_methods/metropolis.py:404: error: "RaveledVars" has no attribute "shape"
pymc/step_methods/metropolis.py:496: error: Signature of "astep" incompatible with supertype "ArrayStep"
pymc/step_methods/metropolis.py:496: note:      Superclass:
pymc/step_methods/metropolis.py:496: note:          def [StatsType] astep(self, apoint: RaveledVars, point: Dict[str, ndarray[Any, Any]], *args: Any) -> Union[RaveledVars, Tuple[RaveledVars, StatsType]]
pymc/step_methods/metropolis.py:496: note:      Subclass:
pymc/step_methods/metropolis.py:496: note:          def astep(self, q0: RaveledVars, logp: Callable[[RaveledVars], ndarray[Any, Any]]) -> RaveledVars
pymc/step_methods/metropolis.py:948: error: Need type annotation for "_history" (hint: "_history: List[<type>] = ...")
pymc/step_methods/mlda.py:351: error: Optional[...] must have exactly one type argument
pymc/step_methods/mlda.py:353: error: Optional[...] must have exactly one type argument
pymc/step_methods/mlda.py:357: error: Optional[...] must have exactly one type argument
pymc/step_methods/mlda.py:360: error: Optional[...] must have exactly one type argument
pymc/step_methods/mlda.py:361: error: Incompatible default for argument "subsampling_rates" (default has type "int", argument has type "List[int]")
pymc/step_methods/mlda.py:411: error: "Model" has no attribute "Q"
pymc/step_methods/mlda.py:446: error: "Model" has no attribute "mu_B"
pymc/step_methods/mlda.py:447: error: "Model" has no attribute "Sigma_B"
pymc/step_methods/mlda.py:460: error: "Model" has no attribute "mu_B"
pymc/step_methods/mlda.py:460: error: "Model" has no attribute "Sigma_B"
pymc/step_methods/mlda.py:486: error: Cannot determine type of "subsampling_rates"
pymc/step_methods/mlda.py:488: error: Cannot determine type of "subsampling_rates"
pymc/step_methods/mlda.py:538: error: Incompatible types in assignment (expression has type "Optional[List[Any]]", variable has type "List[Variable[Any, Any]]")
pymc/step_methods/mlda.py:593: error: Incompatible types in assignment (expression has type "DEMetropolisZMLDA", variable has type "MetropolisMLDA")
pymc/step_methods/mlda.py:609: error: Cannot determine type of "subsampling_rates"
pymc/step_methods/mlda.py:629: error: Incompatible types in assignment (expression has type "MLDA", variable has type "MetropolisMLDA")
pymc/step_methods/mlda.py:685: error: Need type annotation for "Q_diff" (hint: "Q_diff: List[<type>] = ...")
pymc/step_methods/mlda.py:689: error: Need type annotation for "Q_base_full" (hint: "Q_base_full: List[<type>] = ...")
pymc/step_methods/mlda.py:723: error: Incompatible types in assignment (expression has type "int", variable has type "None")
pymc/step_methods/mlda.py:725: error: Incompatible types in assignment (expression has type "int", variable has type "None")
pymc/step_methods/mlda.py:753: error: "Model" has no attribute "Q"
pymc/step_methods/mlda.py:782: error: Incompatible types in assignment (expression has type "MetropolisMLDA", variable has type "MLDA")
pymc/step_methods/mlda.py:790: error: Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "None")
pymc/step_methods/mlda.py:795: error: "None" has no attribute "append"
pymc/step_methods/mlda.py:813: error: Unsupported operand types for + ("List[Dict[str, Any]]" and "None")
pymc/variational/opvi.py:694: error: Need type annotation for "__param_spec__" (hint: "__param_spec__: Dict[<type>, <type>] = ...")
pymc/variational/opvi.py:696: error: Need type annotation for "alias_names"
pymc/variational/opvi.py:697: error: Need type annotation for "__param_registry" (hint: "__param_registry: Dict[<type>, <type>] = ...")
pymc/variational/opvi.py:698: error: Need type annotation for "__name_registry" (hint: "__name_registry: Dict[<type>, <type>] = ...")
pymc/variational/opvi.py:1515: error: Incompatible types in assignment (expression has type "MultiTrace", variable has type "NDArray")
pymc/variational/opvi.py:1519: error: Argument 1 to "to_inference_data" has incompatible type "NDArray"; expected "Optional[MultiTrace]"
pymc/variational/approximations.py:334: error: Incompatible types in assignment (expression has type "Type[MeanFieldGroup]", base class "SingleGroupApproximation" defined the type as "None")
pymc/variational/approximations.py:343: error: Incompatible types in assignment (expression has type "Type[FullRankGroup]", base class "SingleGroupApproximation" defined the type as "None")
pymc/variational/approximations.py:352: error: Incompatible types in assignment (expression has type "Type[EmpiricalGroup]", base class "SingleGroupApproximation" defined the type as "None")
pymc/sampling_jax.py:13: error: Incompatible types in assignment (expression has type "List[str]", variable has type "str")
pymc/sampling_jax.py:14: error: Unsupported operand types for + ("List[str]" and "str")
pymc/sampling_jax.py:71: error: Argument "replace" to "clone_replace" has incompatible type "Dict[SharedVariable, TensorConstant[Any]]"; expected "Union[Iterable[Tuple[Variable[Any, Any], Variable[Any, Any]]], Dict[Variable[Any, Any], Variable[Any, Any]], None]"
pymc/sampling_jax.py:72: error: Incompatible return value type (got "List[Variable[Any, Any]]", expected "List[TensorVariable[Any, Any]]")
pymc/sampling_jax.py:81: error: Argument 1 to "_replace_shared_variables" has incompatible type "Optional[List[TensorVariable[Any, Any]]]"; expected "List[TensorVariable[Any, Any]]"
pymc/sampling_jax.py:93: error: "FunctionGraph" has no attribute "has_destroyers"
pymc/sampling_jax.py:105: error: Unsupported operand type for unary - ("Union[Variable[Any, Any], List[Variable[Any, Any]]]")
pymc/sampling_jax.py:106: error: List item 0 has incompatible type "Union[Variable[Any, Any], List[Variable[Any, Any]]]"; expected "TensorVariable[Any, Any]"
pymc/sampling_jax.py:138: error: Argument "outputs" to "get_jaxified_graph" has incompatible type "Union[Variable[Any, Any], List[Variable[Any, Any]]]"; expected "Optional[List[TensorVariable[Any, Any]]]"
pymc/sampling_jax.py:167: error: List comprehension has incompatible type List[List[ndarray[Any, Any]]]; expected List[Dict[str, ndarray[Any, Any]]]
pymc/sampling_jax.py:169: error: Incompatible types in assignment (expression has type "Dict[str, ndarray[Any, Any]]", variable has type "List[Dict[str, ndarray[Any, Any]]]")
pymc/sampling_jax.py:171: error: List comprehension has incompatible type List[ndarray[Any, dtype[Any]]]; expected List[Dict[str, ndarray[Any, Any]]]
pymc/sampling_jax.py:172: error: Incompatible return value type (got "List[Dict[str, ndarray[Any, Any]]]", expected "Union[ndarray[Any, Any], List[ndarray[Any, Any]]]")
Found 268 errors in 18 files (checked 73 source files)
```

When trying to bump the pre-commit version in https://github.com/pymc-devs/pymc/pull/5995 I got these errors. CC @michaelosthege

--- a/pymc/tests/test_util.py
+++ b/pymc/tests/test_util.py
@@ -14,13 +14,20 @@
 
 import numpy as np
 import pytest
+import xarray
 
 from cachetools import cached
 
 import pymc as pm
 
 from pymc.distributions.transforms import RVTransform
-from pymc.util import UNSET, hash_key, hashable, locally_cachedmethod
+from pymc.util import (
+    UNSET,
+    dataset_to_point_list,
+    hash_key,
+    hashable,
+    locally_cachedmethod,
+)
 
 
 class TestTransformName:
@@ -142,3 +149,18 @@ def test_unset_repr(capsys):
     help(fn)
     captured = capsys.readouterr()
     assert "a=UNSET" in captured.out
+
+
+def test_dataset_to_point_list():
+    ds = xarray.Dataset()
+    ds["A"] = xarray.DataArray([[1, 2, 3]] * 2, dims=("chain", "draw"))
+    pl = dataset_to_point_list(ds)
+    assert isinstance(pl, list)
+    assert len(pl) == 6
+    assert isinstance(pl[0], dict)
+    assert isinstance(pl[0]["A"], np.ndarray)
+
+    # Check that non-str keys are caught
+    ds[3] = xarray.DataArray([1, 2, 3])
+    with pytest.raises(ValueError, match="must be str"):
+        dataset_to_point_list(ds)

--- a/pymc/util.py
+++ b/pymc/util.py
@@ -14,7 +14,7 @@
 
 import functools
 
-from typing import Dict, List, Tuple, Union, cast
+from typing import Dict, Hashable, List, Tuple, Union, cast
 
 import arviz
 import cloudpickle
@@ -232,15 +232,18 @@ def biwrap(wrapper):
 
 
 def dataset_to_point_list(ds: xarray.Dataset) -> List[Dict[str, np.ndarray]]:
+    # All keys of the dataset must be a str
+    for vn in ds.keys():
+        if not isinstance(vn, str):
+            raise ValueError(f"Variable names must be str, but dataset key {vn} is a {type(vn)}.")
     # make dicts
-    points: List[Dict[str, np.ndarray]] = []
-    vn: str
+    points: List[Dict[Hashable, np.ndarray]] = []
     da: "xarray.DataArray"
     for c in ds.chain:
         for d in ds.draw:
             points.append({vn: da.sel(chain=c, draw=d).values for vn, da in ds.items()})
     # use the list of points
-    return points
+    return cast(List[Dict[str, np.ndarray]], points)
 
 
 def chains_and_samples(data: Union[xarray.Dataset, arviz.InferenceData]) -> Tuple[int, int]:

--- a/scripts/run_mypy.py
+++ b/scripts/run_mypy.py
@@ -68,6 +68,7 @@ pymc/step_methods/step_sizes.py
 pymc/tuning/__init__.py
 pymc/tuning/scaling.py
 pymc/tuning/starting.py
+pymc/util.py
 pymc/variational/__init__.py
 pymc/variational/callbacks.py
 pymc/variational/inference.py

